src/main/java/interpreter/PlayerScriptCommand.java

```diff
@@ -0,0 +1,19 @@
+package main.java.interpreter;
+
+
+import main.java.utilties.entities.Player;
+import main.java.utilties.models.Location;
+import main.java.utilties.models.Map;
+
+/*
+*  A class that implements PlayerScriptCommand is added to a list when the drag-and-drop GUI is 'compiled',
+*  and performs the action it specifies in 'void performAction();' when prompted by the game clock.
+* */
+public interface PlayerScriptCommand {
+
+    /*
+    *  This function will be called by the class in order to modify the game in some way.
+    * */
+    void performAction(Map map, Location location, Player player);
```

Comment:
For player commands, I am thinking some actions will require different parameters than Map, Location and Player.  Maybe add a TODO to state we will need to think about this implementation at a later date ? (When we add more advanced commands).

Reply:
Potentially, we could have one parameter for _performAction_ which is a list of the parameters for each scripting function.

`void performAction(ArrayList<Parameter> parameters)`

Reply:
I think we can just pass the 'game' itself into the function, that would probably be easier, since each function is only ever going to modify data in the 'game' class.

Reply:
We will need information specific to each function.  For example, for movement, we will need a location to move to.

Reply:
It might be simpler to make it so we have super specific functions to call then, like moveup, movedown, etc instead of move